src/rust/engine/src/nodes.rs

```diff
@@ -1059,6 +1064,15 @@ impl Node for NodeKey {
       NodeKey::Snapshot(n) => n.run(context).map(|v| v.into()).to_boxed(),
       NodeKey::Task(n) => n.run(context).map(|v| v.into()).to_boxed(),
     }
+    .inspect(move |_: &NodeResult| {
+      let end_timestamp = std::time::SystemTime::now()
+          .duration_since(std::time::SystemTime::UNIX_EPOCH)
+          .unwrap()
+          .as_secs() as f64;
+      let node_name = "Node".to_string();
```

Comment:
`format!("{}", self)` should be a decent representation of the Node

(Note: You may need to do this before the `match self` because `self` won't necessarily exist in the callback)

Reply:
It would be good to use the `should_record_zipkin_spans` flag to "do nothing" here, rather than doing so in the `Session`.

That might look like changing the code at the head of the method to do something like:
```
let node_and_start_time: Option<(String,u64)> = if context.session.should_record_zipkin_spans {
  let start_time = ..;
  let node = format!("{}", ..);
  Some((node, start_time))
} else {
  None
};
```
... and then in the `inspect` block doing an `if let Some((node, start_time)) = node_and_start_time { .. }`.